src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -612,19 +791,105 @@ private SnapshotSummaryModel getSnapshotSummary(UUID snapshotId) {
         snapshotId, snapshotService::retrieveSnapshotSummary);
   }
 
+  @VisibleForTesting
+  DRSObject mergeDRSObjects(List<DRSObject> drsObjects) {
+    DRSObject drsObject =
+        new DRSObject()
+            // Extract singleton values
+            .id(extractUniqueDrsObjectValue(drsObjects, DRSObject::getId))
```

Comment:
This may be more trouble than it's worth, but it looks like `mergeDRSObjects` will throw on the first non-unique value of our singletons.  Any way to collect those conflicts and throw them all at the end?

Reply:
Huh, that's interesting.  lemme see if I can think of a way to handle that